core/templates/dev/head/services/StateRulesStatsService.js

```diff
@@ -58,22 +58,16 @@ oppia.factory('StateRulesStatsService', [
             exploration_id: explorationId,
             visualizations_info: response.data.visualizations_info.map(
               function(vizInfo) {
-                var vizInfoDataWithAddressedInfo = {};
                 if (vizInfo.addressed_info_is_supported) {
-                  vizInfoDataWithAddressedInfo = {
-                    data: vizInfo.data.map(function(vizInfoDatum) {
-                      return Object.assign({
-                        is_addressed: (
-                          AnswerClassificationService
-                            .isClassifiedExplicitlyOrGoesToNewState(
-                              state.name, state,
-                              vizInfoDatum.answer, interactionRulesService))
-                      }, vizInfoDatum);
-                    })
-                  };
+                  vizInfo.data.forEach(function(vizInfoDatum) {
```

Comment:
Hm, we shouldn't be modifying dicts midway, otherwise it's not clear what the keys of the dict are at any given point.

It would be better to use angular.copy() to make a new object (ideally a domain object) and set its fields accordingly.

Reply:
Done, but seems merging/updating is blocked by `oppia.core.controllers.suggestion_test`?